### ISSUE TITLE
snps.aligned.fa consensus is different from snps.bam on TYPE="complex"

### ISSUE THREAD

**Issue by noeldjitro** [tseemann/snippy#266]
Hi Torsten,
Thank you for making this tool

I found an issue when comparing snps.bam (i opened the alignment with tablet), snps.filt.vcf and snps.aligned.fa

The problem occur when when the TYPE in vcf is detected as "complex", and the amount of nucleotide from reference is different from alternatife.

![tablet snps filt vcf and fasta](https://user-images.githubusercontent.com/47404887/52384924-a10d9a00-2ad3-11e9-8b02-51435e055b36.png)

![screenshot from 2019-02-07 13-58-41](https://user-images.githubusercontent.com/47404887/52388041-ae7d5100-2ae0-11e9-86f3-b3eeceb4024f.png)

Can you fix this problem, please?

Thank you
Noel

**Comment by noeldjitro** [comment]
my bad, just re-read the description on snippy page and found out this
.aligned.fa -> A version of the reference but with - at position with depth=0 and N for 0 < depth < --mincov (does not have variants)

**Comment by tseemann** [comment]
What version of Snippy are you using?  
(run `snippy --version`)

`snps.filt.vcf` is not the file that is for the end-user, that file is `snps.vcf`.

It is hard to follow the images, but it seems you have a `GG => TGA` called.
The main problem with indels like this is the ambiguity.
Is it:
```
(1)   (2)
GG-   -GG
TGA . TGA
```

**Comment by noeldjitro** [comment]
The version is 4.3.6

So, in position 74045-74047 (left)
The reference is GGT
The alignment in bam is GAT

In snps.vcf, it is GG --> TGA (right top)

However, when I opened the snps.align.fa, the result is different. (right bottom)


![sample](https://user-images.githubusercontent.com/47404887/61424970-b1665280-a958-11e9-9373-da4f429b0563.jpg)
